Revit_Core_Engine/Create/View/ViewReflectedCeilingPlan.cs

```diff
@@ -39,18 +39,18 @@ public static partial class Create
         [Input("viewName", "Optional, name of the new view.")]
         [Input("cropBox", "Optional, the crop region to attempt to apply to the newly created view.")]
         [Input("viewTemplateId", "Optional, the View Template Id to be applied in the view.")]
-        [Input("viewDetailLevel", "Optional, the Detail Level of the view.")]        
-        [Output("viewReflectedCeilingPlan", "The new view.")]        
+        [Input("viewDetailLevel", "Optional, the Detail Level of the view.")]
+        [Output("viewReflectedCeilingPlan", "The new view.")]
         public static View ViewReflectedCeilingPlan(this Document document, Autodesk.Revit.DB.Level level, string viewName = null, CurveLoop cropBox = null, ElementId viewTemplateId = null, ViewDetailLevel viewDetailLevel = ViewDetailLevel.Coarse)
         {
             View result = null;
 
             ViewFamilyType vft = Query.ViewFamilyType(document, ViewFamily.CeilingPlan);
 
             result = Autodesk.Revit.DB.ViewPlan.Create(document, vft.Id, level.Id);
-            
+
             Modify.SetViewDetailLevel(result, viewDetailLevel);
-            
+
             if (cropBox != null)
             {
                 try
```

Comment:
You can check the view's `ViewTemplateId` property and only apply the crop box if the template is in use. Then, the `RecordWarning` in this `try catch` can only mention "...the crop box is a valid geometry..." to make it more specific for the user.

Reply:
As discussed, view templates and cropbox settings are not connected, so I will change the warning message.